### DISCUSSION TITLE
Section-properties is not running for me using Python 3.11.6

### DISCUSSION THREAD

**Discussion by aladdin100001** [robbievanleeuwen/section-properties#347] [Q&A] [answered]
I am not sure what is the problem and I hope I can get some guidance. I downloaded section-properties four months ago and it was working for me using python 3.9 (I am using visual studio IDE). I downloaded the latest section-properties yesterday and it is not working for me using python 3.9 or 3.11. Does anyone experience the same problem?

**Comment by robbievanleeuwen** [comment]
Hi @aladdin100001,

I'm wondering if you could provide some further information to allow us to help you out. Could you please answer the following questions:

- When you say it is not working for you, what is the error message/traceback you are getting? It's very difficult yo debug your issue without this information.
- How are you trying to install `sectionproperties`? 
  - Are you using `pip install sectionproperties`?
  - Have you tried installing into a fresh virtual environment?
  - Have you tried the above, pinning `sectionproperties` to the latest version, i.e. `pip install sectionproperties==3.1.1`
- What operating system are you using?

**Reply by aladdin100001** [reply]
Thank you for replying. I was waiting for Australia to wake up:)
This is the output i get:
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\python.exe'. Symbols loaded.
'python.exe' (Win32): Loaded 'C:\Windows\System32\ntdll.dll'. 
'python.exe' (Win32): Loaded 'C:\Windows\System32\hmpalert.dll'. 
'python.exe' (Win32): Loaded 'C:\Windows\System32\kernel32.dll'. 
'python.exe' (Win32): Loaded 'C:\Windows\System32\KernelBase.dll'. 
'python.exe' (Win32): Loaded 'C:\Windows\System32\ucrtbase.dll'. 
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\python311.dll'. Symbols loaded.
'python.exe' (Win32): Loaded 'C:\Windows\System32\ws2_32.dll'. 
'python.exe' (Win32): Loaded 'C:\Windows\System32\rpcrt4.dll'. 
'python.exe' (Win32): Loaded 'C:\Windows\System32\advapi32.dll'. 
'python.exe' (Win32): Loaded 'C:\Windows\System32\msvcrt.dll'. 
'python.exe' (Win32): Loaded 'C:\Windows\System32\sechost.dll'. 
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\vcruntime140.dll'. 
'python.exe' (Win32): Loaded 'C:\Windows\System32\version.dll'. 
'python.exe' (Win32): Loaded 'C:\Windows\System32\bcrypt.dll'. 
'python.exe' (Win32): Loaded 'C:\Windows\System32\bcryptprimitives.dll'. 
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\python3.dll'. 
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\DLLs\_bz2.pyd'. Symbols loaded.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\DLLs\_lzma.pyd'. Symbols loaded.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numpy\core\_multiarray_umath.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numpy.libs\libopenblas64__v0.3.23-293-gc2f4bdbb-gcc_10_3_0-2bde3a66a51006b2b53eb373ff767a3f.dll'. 
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numpy\core\_multiarray_tests.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\DLLs\_ctypes.pyd'. Symbols loaded.
'python.exe' (Win32): Loaded 'C:\Windows\System32\ole32.dll'. 
'python.exe' (Win32): Loaded 'C:\Windows\System32\msvcp_win.dll'. 
'python.exe' (Win32): Loaded 'C:\Windows\System32\gdi32.dll'. 
'python.exe' (Win32): Loaded 'C:\Windows\System32\win32u.dll'. 
'python.exe' (Win32): Loaded 'C:\Windows\System32\gdi32full.dll'. 
'python.exe' (Win32): Loaded 'C:\Windows\System32\user32.dll'. 
'python.exe' (Win32): Loaded 'C:\Windows\System32\combase.dll'. 
'python.exe' (Win32): Loaded 'C:\Windows\System32\oleaut32.dll'. 
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\DLLs\libffi-8.dll'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Windows\System32\imm32.dll'. 
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numpy\linalg\_umath_linalg.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numpy\fft\_pocketfft_internal.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numpy\random\mtrand.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numpy\random\bit_generator.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numpy\random\_common.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\DLLs\_hashlib.pyd'. Symbols loaded.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\DLLs\libcrypto-3.dll'. Symbols loaded.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numpy\random\_bounded_integers.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numpy\random\_mt19937.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numpy\random\_philox.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numpy\random\_pcg64.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numpy\random\_sfc64.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numpy\random\_generator.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\matplotlib\_c_internal_utils.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Windows\System32\shell32.dll'. 
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\PIL\_imaging.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\matplotlib\_path.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\matplotlib.libs\msvcp140-7a5216ee7dbbe0be4badcb1661870433.dll'. 
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\matplotlib\ft2font.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\kiwisolver\_cext.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Windows\System32\msvcp140.dll'. 
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\vcruntime140_1.dll'. 
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\DLLs\unicodedata.pyd'. Symbols loaded.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\matplotlib\_image.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\DLLs\_uuid.pyd'. Symbols loaded.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\DLLs\_decimal.pyd'. Symbols loaded.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\DLLs\_queue.pyd'. Symbols loaded.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\shapely\lib.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\shapely.libs\geos_c-69e7a24742c4fc930f28f9e1af49d3da.dll'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\shapely.libs\geos-0b3f0497f548b611a538169737f35281.dll'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\shapely.libs\msvcp140-984b5b930a73989563f66aa5ca23a5b8.dll'. 
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\shapely\_geometry_helpers.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\shapely\_geos.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\DLLs\_socket.pyd'. Symbols loaded.
'python.exe' (Win32): Loaded 'C:\Windows\System32\IPHLPAPI.DLL'. 
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\DLLs\select.pyd'. Symbols loaded.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\_lib\_ccallback_c.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\sparse\_sparsetools.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\sparse\_csparsetools.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\sparse\linalg\_isolve\_iterative.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy.libs\libopenblas_v0.3.20-571-g3dec11c6-gcc_10_3_0-c2315440d6b6cef5037bad648efc8c59.dll'. 
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\linalg\_fblas.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\linalg\_flapack.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\linalg\_cythonized_array_utils.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\linalg\cython_lapack.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\linalg\_solve_toeplitz.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\linalg\_decomp_lu_cython.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\linalg\_matfuncs_sqrtm_triu.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\linalg\_matfuncs_expm.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\linalg\cython_blas.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\linalg\_decomp_update.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\linalg\_flinalg.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\sparse\linalg\_dsolve\_superlu.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\sparse\linalg\_eigen\arpack\_arpack.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\sparse\csgraph\_shortest_path.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\sparse\csgraph\_tools.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\sparse\csgraph\_traversal.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\sparse\csgraph\_min_spanning_tree.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\sparse\csgraph\_flow.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\sparse\csgraph\_matching.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\scipy\sparse\csgraph\_reordering.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\llvmlite\binding\llvmlite.dll'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numba\core\typeconv\_typeconv.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numba\_helperlib.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numba\_dynfunc.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numba\_dispatcher.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numba\_devicearray.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numba\core\runtime\_nrt_python.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numba\np\ufunc\_internal.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numba\experimental\jitclass\_box.cp311-win_amd64.pyd'. Module was built without symbols.


and then I get a black console with nothing in it. it stays on forever.

yes I used pip for a new environment of python 3.11 and yes i have sectionproperties 3.1.1.
I am running windows 11. I use microsoft visual studio as my IDE (I assume it is not the problem since it was running before).

**Reply by robbievanleeuwen** [reply]
Thanks @aladdin100001,

I'm not a big windows user so I'm surprised to see all this output! Is this what you see every time you run python? I can't see `sectionproperties` in this output so am doubtful it's related to this program in particular. 

Potentially an issue with running python through your IDE? Have you tried running in a standalone command prompt? I've had success with windows terminal preview in the past and using python's virtual environments rather than VS virtual environments.

The other thing you could try is installing `sectionproperties=3.0.2` in a fresh environment. The last line in that output in `numba` which was added in version `3.1.0` and maybe you are having trouble installing `numba` on your computer?

**Reply by aladdin100001** [reply]
Ok, I just confirmed it is working for sectionproperties==3.0.2 and python3.11 but NOT for sectionproperties 3.1.1.
Maybe it is irrelevant but it shows bunch of 
![image](https://github.com/robbievanleeuwen/section-properties/assets/148366612/b2106f30-e1db-45aa-9ba3-4e79239f86e8)

**Reply by aladdin100001** [reply]
I also tried two different machines for python 3.11 and sectionproperties 3.1.1, no luck.

**Reply by robbievanleeuwen** [reply]
> Maybe it is irrelevant but it shows bunch of

I think this is just your IDE - the testing directory was restructured in `3.1.0` but this has nothing to do with the `sectionproperties` package. The tests are only relevant if you want to develop and contribute to `sectionproperties`, they are not important if you are just trying to use it. This is another reason I don't like Visual Studio and prefer VS Code 😆 

The main difference between `3.0.2` and `3.1.0`/`3.1.1` is `numba` so I suspect you are having trouble installing numba on your computer. You could confirm this by just trying `pip install numba` in a fresh environment and seeing if you get the same issue.

**Reply by robbievanleeuwen** [reply]
> I also tried two different machines for python 3.11 and sectionproperties 3.1.1, no luck.

We run [tests](https://github.com/robbievanleeuwen/section-properties/actions/runs/6527201090/job/17721737267) for `sectionproperties` on a windows virtual machine and do not have issues so I suspect there is something conflicting with your python installation or the way you have your virtual environments set up.

**Reply by aladdin100001** [reply]
I just ran sectionproperties 3.1.1 from the terminal directly, it did not work. sectionproperties3.0.2 from the terminal directly worked.
Numba is installed :
----- Installing 'pip install numba' -----
Requirement already satisfied: pip in c:\program files\python311\lib\site-packages (23.3)
Requirement already satisfied: install in c:\program files\python311\lib\site-packages (1.3.5)
Requirement already satisfied: numba in c:\program files\python311\lib\site-packages (0.58.1)
Requirement already satisfied: llvmlite<0.42,>=0.41.0dev0 in c:\program files\python311\lib\site-packages (from numba) (0.41.1)
Requirement already satisfied: numpy<1.27,>=1.22 in c:\program files\python311\lib\site-packages (from numba) (1.26.1)
----- Successfully installed 'pip install numba' -----

**Reply by robbievanleeuwen** [reply]
> I just ran sectionproperties 3.1.1 from the terminal directly, it did not work

Do you get the same output as above? When you say you ran do you mean it has installed already? Or you cannot install?

**Reply by aladdin100001** [reply]
i mean sectionproperties 3.0.2 ran a full script with results and everything. sectionproperties3.1.1 gives a blank black console screen. Both used python 3.11.

**Reply by robbievanleeuwen** [reply]
Sorry @aladdin100001 I thought your issue was that you couldn't install `sectionproperties` 3.1.1. Can you confirm the installation works fine? Could you also place the script you are trying to run here so I can see what you are trying to do? Please appreciate that it is hard to help you without all the information.

**Reply by aladdin100001** [reply]
I understand, I am trying to give you all the information I have. I really appreciate your time too.
So the installation is completed successfully for sectionproperties3.1.1. When i try to run a script/example, it gives a blank console.
The example i am trying to run is from the documentation (the first one). It does run/works using 3.0.2 but not 3.1.1.

**Reply by robbievanleeuwen** [reply]
Thanks @aladdin100001 - confirm [this](https://sectionproperties.readthedocs.io/en/stable/examples/geometry/geometry_coordinates.html) is the example you are trying?

**Reply by robbievanleeuwen** [reply]
Can you run this in the terminal and tell me what it outputs?

```sectionproperties --version```

**Reply by aladdin100001** [reply]
yes it is

**Reply by aladdin100001** [reply]
C:\Users\me>sectionproperties --version
sectionproperties, version 3.1.1

**Reply by robbievanleeuwen** [reply]
Could you also try the below simple script without any plotting? 

Here is the script:

```python
from sectionproperties.pre.library import rectangular_section
from sectionproperties.analysis import Section


box = rectangular_section(100, 100)
box.create_mesh(0)
sec = Section(box)
sec.calculate_geometric_properties()
sec.display_results()
```

**Reply by aladdin100001** [reply]
3.1.1 gives the blank screen.
3.0.2 gives
     Section Properties
┏━━━━━━━━━━━┳━━━━━━━━━━━━━━┓
┃ Property  ┃        Value ┃
┡━━━━━━━━━━━╇━━━━━━━━━━━━━━┩
│ area      │ 1.000000e+04 │
│ perimeter │ 4.000000e+02 │
│ qx        │ 5.000000e+05 │
│ qy        │ 5.000000e+05 │
│ ixx_g     │ 3.333333e+07 │
│ iyy_g     │ 3.333333e+07 │
│ ixy_g     │ 2.500000e+07 │
│ cx        │ 5.000000e+01 │
│ cy        │ 5.000000e+01 │
│ ixx_c     │ 8.333333e+06 │
│ iyy_c     │ 8.333333e+06 │
│ ixy_c     │ 0.000000e+00 │
│ zxx+      │ 1.666667e+05 │
│ zxx-      │ 1.666667e+05 │
│ zyy+      │ 1.666667e+05 │
│ zyy-      │ 1.666667e+05 │
│ rx        │ 2.886751e+01 │
│ ry        │ 2.886751e+01 │
│ i11_c     │ 8.333333e+06 │
│ i22_c     │ 8.333333e+06 │
│ phi       │ 0.000000e+00 │
│ z11+      │ 1.666667e+05 │
│ z11-      │ 1.666667e+05 │
│ z22+      │ 1.666667e+05 │
│ z22-      │ 1.666667e+05 │
│ r11       │ 2.886751e+01 │
│ r22       │ 2.886751e+01 │
└───────────┴──────────────┘

**Reply by robbievanleeuwen** [reply]
This is very mysterious!! Can you run an even simpler script?

```python
from sectionproperties.pre.library import rectangular_section


box = rectangular_section(100, 100)
```

**Reply by aladdin100001** [reply]
The output always stops at numba:
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numba\core\typeconv\_typeconv.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numba\_helperlib.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numba\_dynfunc.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numba\_dispatcher.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numba\_devicearray.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numba\core\runtime\_nrt_python.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numba\np\ufunc\_internal.cp311-win_amd64.pyd'. Module was built without symbols.
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numba\experimental\jitclass\_box.cp311-win_amd64.pyd'. Module was built without symbols.

**Reply by aladdin100001** [reply]
same it stopped after this line:
'python.exe' (Win32): Loaded 'C:\Program Files\Python311\Lib\site-packages\numba\experimental\jitclass_box.cp311-win_amd64.pyd'. Module was built without symbols.

**Reply by robbievanleeuwen** [reply]
Ok thanks @aladdin100001, for now I would recommend just using `sectionproperties==3.0.2`. The only difference from a usability point of view is the addition of `numba` and the optional `pypardiso` which is supposed to improve performance. Everything else should be identical.

I will try to reproduce your issue on a windows computer later this afternoon and see if I can figure out what's going on.

**Reply by aladdin100001** [reply]
Thank you for your time and a great application

**Reply by robbievanleeuwen** [reply]
One last thing, could you paste the output of ``numba -s`` here?

**Reply by robbievanleeuwen** [reply]
Also do you have same issues running different versions of python with `sectionproperties` 3.1.1? E.g. python 3.11.4, 3.10 or 3.9?

**Reply by aladdin100001** [reply]
i did but I am not sure if this is what you are looking for (maybe it is a different parameter in windows), anyway this is what i am getting:
#!C:\Program Files\Python311\python.exe
# -*- coding: UTF-8 -*-
from __future__ import print_function, division, absolute_import

from numba.misc.numba_entry import main

if __name__ == "__main__":
    main()

**Reply by aladdin100001** [reply]
not sure why the font change when i clicked "Reply"

**Reply by aladdin100001** [reply]
I tried sectioproperties 3.1.1 with python 3.9 and I had the same problem

**Reply by robbievanleeuwen** [reply]
This is just from the numba docs to test the installation - maybe try this: ``python -m numba -s``

**Reply by aladdin100001** [reply]
System info:
--------------------------------------------------------------------------------
__Time Stamp__
Report started (local time)                   : 2023-10-18 20:40:29.241092
UTC start time                                : 2023-10-19 01:40:29.241092
Running time (s)                              : 2.732884

__Hardware Information__
Machine                                       : AMD64
CPU Name                                      : alderlake
CPU Count                                     : 16
Number of accessible CPUs                     : ?
List of accessible CPUs cores                 : ?
CFS Restrictions (CPUs worth of runtime)      : None

CPU Features                                  : 64bit adx aes avx avx2 avxvnni bmi
                                                bmi2 clflushopt clwb cmov crc32
                                                cx16 cx8 f16c fma fsgsbase fxsr
                                                gfni hreset invpcid lzcnt mmx
                                                movbe movdir64b movdiri pclmul
                                                popcnt prfchw ptwrite rdpid rdrnd
                                                rdseed sahf serialize sha shstk
                                                sse sse2 sse3 sse4.1 sse4.2 ssse3
                                                vaes vpclmulqdq waitpkg xsave
                                                xsavec xsaveopt xsaves

Memory Total (MB)                             : 54184
Memory Available (MB)                         : 26759

__OS Information__
Platform Name                                 : Windows-10-10.0.22621-SP0
Platform Release                              : 10
OS Name                                       : Windows
OS Version                                    : 10.0.22621
OS Specific Version                           : 10 10.0.22621 SP0 Multiprocessor Free
Libc Version                                  : ?

__Python Information__
Python Compiler                               : MSC v.1935 64 bit (AMD64)
Python Implementation                         : CPython
Python Version                                : 3.11.6
Python Locale                                 : en_US.cp65001

__Numba Toolchain Versions__
Numba Version                                 : 0.58.1
llvmlite Version                              : 0.41.1

__LLVM Information__
LLVM Version                                  : 14.0.6

__CUDA Information__
CUDA Device Initialized                       : False
CUDA Driver Version                           : ?
CUDA Runtime Version                          : ?
CUDA NVIDIA Bindings Available                : ?
CUDA NVIDIA Bindings In Use                   : ?
CUDA Minor Version Compatibility Available    : ?
CUDA Minor Version Compatibility Needed       : ?
CUDA Minor Version Compatibility In Use       : ?
CUDA Detect Output:
None
CUDA Libraries Test Output:
None

__NumPy Information__
NumPy Version                                 : 1.26.1
NumPy Supported SIMD features                 : ('MMX', 'SSE', 'SSE2', 'SSE3', 'SSSE3', 'SSE41', 'POPCNT', 'SSE42', 'AVX', 'F16C', 'FMA3', 'AVX2')
NumPy Supported SIMD dispatch                 : ('SSSE3', 'SSE41', 'POPCNT', 'SSE42', 'AVX', 'F16C', 'FMA3', 'AVX2', 'AVX512F', 'AVX512CD', 'AVX512_SKX', 'AVX512_CLX', 'AVX512_CNL', 'AVX512_ICL')
NumPy Supported SIMD baseline                 : ('SSE', 'SSE2', 'SSE3')
NumPy AVX512_SKX support detected             : False

__SVML Information__
SVML State, config.USING_SVML                 : False
SVML Library Loaded                           : False
llvmlite Using SVML Patched LLVM              : True
SVML Operational                              : False

__Threading Layer Information__
TBB Threading Layer Available                 : False
+--> Disabled due to Unknown import problem.
OpenMP Threading Layer Available              : True
+-->Vendor: MS
Workqueue Threading Layer Available           : True
+-->Workqueue imported successfully.

__Numba Environment Variable Information__
None found.

__Conda Information__
Conda not available.

__Installed Packages__
Package            Version
------------------ --------
click              8.1.7
colorama           0.4.6
concreteproperties 0.6.1
contourpy          1.1.1
cycler             0.12.1
fonttools          4.43.1
install            1.3.5
kiwisolver         1.4.5
llvmlite           0.41.1
markdown-it-py     3.0.0
matplotlib         3.8.0
mdurl              0.1.2
more-itertools     10.1.0
numba              0.58.1
numpy              1.26.1
packaging          23.2
Pillow             10.1.0
pip                23.3
Pygments           2.16.1
pyparsing          3.1.1
python-dateutil    2.8.2
rich               13.6.0
scipy              1.11.3
sectionproperties  3.1.1
setuptools         65.5.0
shapely            2.0.2
six                1.16.0
triangle           20230923

No errors reported.


__Warning log__
Warning (cuda): CUDA driver library cannot be found or no CUDA enabled devices are present.
Exception class: <class 'numba.cuda.cudadrv.error.CudaSupportError'>
Warning: Conda not available.
 Error was [WinError 2] The system cannot find the file specified

Warning (psutil): psutil cannot be imported. For more accuracy, consider installing it.
--------------------------------------------------------------------------------
If requested, please copy and paste the information between
the dashed (----) lines, or from a given specific section as
appropriate.

=============================================================
IMPORTANT: Please ensure that you are happy with sharing the
contents of the information present, any information that you
wish to keep private you should remove before sharing.
=============================================================

**Reply by robbievanleeuwen** [reply]
Thanks @aladdin100001, it seems like your numba should be working fine but clearly something is not working as it should! I suggest working with `sectionproperties==3.0.2` for now. I will work on making `numba` an optional dependency in the future and hopefully this works for you and other users that might have this problem. I have also tried with a windows computer and couldn't reproduce your problem.

Potentially the last thing you could try is installing different versions of numba, e.g. 0.58.0, 0.57.1, *after* installing `sectionproperties`, to see if any of these will work. You will get a `pip` error but they should still install.

**Reply by robbievanleeuwen** [reply, marked as answer]
@aladdin100001 I've introduced `numba` as an optional dependency now. I haven't made a new release but you can test out the `master` branch if you'd like. After creating a fresh virtual environment:

```shell
git clone https://github.com/robbievanleeuwen/section-properties.git
cd section-properties
pip install .
```

Hopefully this works for you! Note depending on which terminal you use `cd` (change directory) may need to be something else.

**Reply by aladdin100001** [reply]
Thank you, I will do. Would you like to mark this thread as answered ,close it?

**Reply by robbievanleeuwen** [reply]
Let's wait until you confirm it's working 😄  Hopefully it's not another issue!

**Reply by aladdin100001** [reply]
that works for me:) thanks
I will keep you in the loop if i find something about my numba problem
Thank you again
